src/java.base/linux/classes/jdk/internal/platform/CgroupSubsystemFactory.java

```diff
@@ -108,12 +108,12 @@ static CgroupMetrics create() {
         Map<String, CgroupInfo> infos = result.getInfos();
         if (result.isCgroupV2()) {
             // For unified it doesn't matter which controller we pick.
-            CgroupInfo anyController = infos.get(MEMORY_CTRL);
+            CgroupInfo anyController = infos.values().iterator().next();
             CgroupSubsystem subsystem = CgroupV2Subsystem.getInstance(anyController);
-            return subsystem != null ? new CgroupMetrics(subsystem) : null;
+            return new CgroupMetrics(subsystem);
         } else {
             CgroupV1Subsystem subsystem = CgroupV1Subsystem.getInstance(infos);
-            return subsystem != null ? new CgroupV1MetricsImpl(subsystem) : null;
+            return new CgroupV1MetricsImpl(subsystem);
```

Comment:
This shouldn't be changed because the current implementation of `CgroupV1Subsystem.getInstance(infos)` has a path that returns null.

Maybe that's impossible, because when we call `CgroupV1Subsystem.getInstance`, we must have at least one v1 subsystem in `infos`. However, that's not related to this issue.  Please fix that in a separate RFE. For example, `CgroupV1Subsystem.getInstance(infos)`  can be changed to throw an exception instead if return null.

Reply:
Fine by me; done.